server/src/main/java/com/lunchtime/controllers/RestaurantResource.java

```diff
@@ -0,0 +1,97 @@
+package com.lunchtime.controllers;
+
+import com.lunchtime.models.Restaurant;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.lunchtime.service.RestaurantService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/restaurants")
+public class RestaurantResource {
+
+    @Autowired
+    private RestaurantService restaurantService;
```

Comment:
Since Spring 4.3 it is recommended to avoid using `@Autowired` annotation but use **final** field (`private final RestaurantService restaurantService` in our case) and injecting by constructor.

```
public RestaurantResource (RestaurantService restaurantService){
        this.restaurantService= restaurantService;
    }
```

Reply:
Thanks, fixed